### DISCUSSION TITLE
[webpack-dev-middleware] [Error:   × TypeError: Cannot read properties of undefined (reading 'set')

### DISCUSSION THREAD

**Discussion by ThomasXing** [web-infra-dev/rsbuild#3445] [Q&A]
e> [webpack-dev-middleware] [Error:   × TypeError: Cannot read properties of undefined (reading 'set')
<e>   │     at Object.fn (/Users/thomasxing/workspace/admin/web/node_modules/.pnpm/webpack@5.94.0/node_modules/webpack/lib/container/ContainerReferencePlugin.js:84:37)
<e>   │     at SyncHook.callAsyncStageRange (/Users/thomasxing/workspace/admin/web/node_modules/.pnpm/@rspack+lite-tapable@1.0.0/node_modules/@rspack/lite-tapable/dist/index.js:240:21)
<e>   │     at SyncHook.callStageRange (/Users/thomasxing/workspace/admin/web/node_modules/.pnpm/@rspack+lite-tapable@1.0.0/node_modules/@rspack/lite-tapable/dist/index.js:259:14)
<e>   │     at QueriedHook.call (/Users/thomasxing/workspace/admin/web/node_modules/.pnpm/@rspack+lite-tapable@1.0.0/node_modules/@rspack/lite-tapable/dist/index.js:218:26)
<e>   │     at /Users/thomasxing/workspace/admin/web/node_modules/.pnpm/@rspack+core@1.0.4_@swc+helpers@0.5.13/node_modules/@rspack/core/dist/Compiler.js:572:244
<e>   │     at last.function (/Users/thomasxing/workspace/admin/web/node_modules/.pnpm/@rspack+core@1.0.4_@swc+helpers@0.5.13/node_modules/@rspack/core/dist/Compiler.js:862:28)
<e>   │ 
<e> ] {
<e>   code: 'GenericFailure'
<e> }

还没有对dev-middleware 做相关配置 不知道为啥这块报错

**Comment by 9aoy** [comment]
>(/Users/thomasxing/workspace/admin/web/node_modules/.pnpm/webpack@5.94.0/node_modules/webpack/lib/container/ContainerReferencePlugin.js:84:37)

use rspack plugin instead. https://rspack.dev/plugins/webpack/internal-plugins#containerreferenceplugin

**Reply by 9aoy** [reply]
Please provide a reproduction repo.

**Reply by ThomasXing** [reply]
× Module build failed:
  ├─▶   ×   × Expected '>', got 'title'
  │     │      ╭─[/Users/thomasxing/workspace/admin/web/src/views/financialCenter/settlementList/settlementAdd.vue:137:1]
  │     │  134 │       }
  │     │  135 │     },
  │     │  136 │     renderFunc(h, option) {
  │     │  137 │       return <span title={option.label}>{option.label}</span>
  │     │      ·                    ─────
  │     │  138 │     },
  │     │  139 │     // 失败弹窗返回参数
  │     │  140 │     getUpload(num) {
  │     │      ╰────
  │     │
  │   
  ╰─▶ Syntax Error

**Reply by ThomasXing** [reply]
刚才的报错是需要@rsbuild/plugin-vue2 插件， 麻烦看下这种render的渲染 需要更换哪个插件

**Reply by ThomasXing** [reply]
![image](https://github.com/user-attachments/assets/a77a5374-3d6d-4934-bdf1-00d6ac5c3c87)
element-ui 里的组件库里的 el-tree 节点render渲染

**Reply by 9aoy** [reply]
提供下可复现的 demo，只看代码片段看不出来

**Comment by liyincode** [comment]
You should have encountered an error in the renderContent method due to the use of JSX syntax. Install the [rsbuild-plugin-vue2-jsx](https://github.com/rspack-contrib/rsbuild-plugin-vue2-jsx) plugin and add lang="jsx" to the script section, like `<script lang="jsx">`. Then try again.
